add some notes about pinning python dependencies in CONTRIBUTING.rst (#198)
OCA/odoo-community.org: website/Contribution/CONTRIBUTING.rst

        'installable': True,
    }

Regarding python package pinning, it is a responsibility of the integrator delivering the end-project.

* a module should never place exact pins on their dependencies (whether they are also Odoo dependencies or not);
* a module may require a lower bound (e.g. external_dependency_python>1.4) if they depend on recent features of a library (but not higher than necessary);
* a module may exceptionally place an upper bound if they know they are incompatible with recent versions of a library AND it is not feasible to fix the compatibility issue.

If too strict bounds or pinning is done in module, then integrators quickly land in dependency hell with unresolvable depency conflicts.

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions), but Odoo does work perfectly well with more recent versions, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Reviewer: ```suggestion
Note that versions in Odoo's requirement.txt are a recommendation (known good versions used in some Debian and/or Ubuntu versions), but Odoo does work perfectly well with more recent versions, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.
```

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions), but Odoo does work perfectly well with more recent versions, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Reviewer 2: Don't be so categorical. In fact, I don't feel comfortable advising about using latest library versions. Some libraries like `werkzeug` and other that are monkey-patched can't be changed from the fixed version ones.

```suggestion
Note that versions in Odoo's requirement.txt are a recommendation (known good versions), but Odoo may work well with more recent versions, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.
```

Reviewer: Agreed.

We could say "Odoo works well with more recent version of many libraries".

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions used in some Debian and/or Ubuntu versions), but Odoo may work well with more recent versions of many libraries, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Reviewer 2: ```suggestion
Note that versions in Odoo's requirements.txt are the tested ones (bundled versions used in the latest Debian stable version), but Odoo may work well with more recent versions of many libraries, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.
```

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions used in some Debian and/or Ubuntu versions), but Odoo may work well with more recent versions of many libraries, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Author: > we should stick to the pinned versions as much as possible.

Wow; I had the feeling there was shared agreement on this.

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions used in some Debian and/or Ubuntu versions), but Odoo may work well with more recent versions of many libraries, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Reviewer 2: Well, I already said in https://github.com/OCA/odoo-community.org/pull/198#discussion_r1928523413 that we shouldn't advise the use of latest versions.

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions used in some Debian and/or Ubuntu versions), but Odoo may work well with more recent versions of many libraries, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Reviewer 3: what about "it may happen" instead "frequently happens" so it looks more neutral. frequently is a bit subjective and depends also about the point of view like: it might be frequent for us developpers who specialize into delivering cutting edge technologies but it's probably not so frequent for the average OCA module user who will like something easy to install that just work out of the box with the standard Odoo dependencies.

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions used in some Debian and/or Ubuntu versions), but Odoo may work well with more recent versions of many libraries, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Author: It's not an advice to use lastest version, it's about to not pin version without a good reason in oca.

I will %s/frequently/may as suggested by @rvalyi to be more neutral.

Again, the recommandation here, in this paragraph, is not about use modern or outdated dependencies.

> Note that versions in Odoo's requirement.txt are a recommendation (known good versions used in some Debian and/or Ubuntu versions), but Odoo may work well with more recent versions of many libraries, and it frequently happens that one needs to use more recent versions of the Odoo requirements to be compatible with other modern libraries.

Reviewer 4: Thanks, I like @rvalyi suggestion.
